curated_transformers/models/module.py

```diff
@@ -101,23 +101,31 @@ class EncoderModule(Module):
     @abstractmethod
     def forward(
         self,
-        input_ids: Tensor,
+        piece_ids: Tensor,
```

Comment:
Probably for a different PR, but we should check and normalize where we want to pass positional arguments and where keyword-only arguments work better. I don't think we currently follow a standard protocol that spans the entire codebase.

Reply:
My main rule has been when there can be confusion between arguments because they take the same type. But we haven't really nailed that down much. E.g. if you have:

```
def foo(a: Tensor, b: Tensor):
```

then it's really easy to mix up the arguments and type checking won't help. So, we could make them keyword arguments:

```
def foo(*, a: Tensor, b: Tensor):
```

which is nice and clean. However,

```
def foo(a: Tensor, *, b: Tensor):
```

would probably be enough, since `a l is disambiguated by virtue of needing to specify `b`. However, I don's like this, _except_ in cases where it `a` would very obviously the first, most important argument (e.g. piece ids for a transformer). Especially in the case where the other arguments have default values, it would be quite jarring having to specify the name of that first argument.

I guess there is sort of another concern when types are different, that the argument list may get too long to use positional arguments. But it's hard to come up with a cut-off.